### ISSUE TITLE
Problem with importing erfa and astropy or photutils in iPython/Spyder

### ISSUE THREAD

**Issue by kbagnall-quera** [liberfa/pyerfa#92]
Hi,
I'm a newbie - sorry for my inexperience. I'm running iPython inside Spyder and trying to use astropy and photutils. I installed pyerfa, astropy, and photutils inside iPython using '!pip install pyerfa' but I get the following error when I try to import erfa.

import erfa
[autoreload of erfa._version failed: Traceback (most recent call last):
  File "C:\Users\kbagnall\AppData\Local\Programs\Spyder\pkgs\IPython\extensions\autoreload.py", line 245, in check
    superreload(m, reload, self.old_objects)
  File "C:\Users\kbagnall\AppData\Local\Programs\Spyder\pkgs\IPython\extensions\autoreload.py", line 394, in superreload
    module = reload(module)
  File "imp.py", line 314, in reload
  File "importlib\__init__.py", line 159, in reload
ImportError: parent 'erfa' not in sys.modules
]
Traceback (most recent call last):

  File "C:\Users\kbagnall\AppData\Local\Temp\ipykernel_14788\3657507462.py", line 1, in <cell line: 1>
    import erfa

  File "C:\Users\kbagnall\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\erfa\__init__.py", line 5, in <module>
    from .version import version as __version__  # noqa

  File "C:\Users\kbagnall\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\erfa\version.py", line 25, in <module>
    from . import ufunc

ImportError: cannot import name 'ufunc' from partially initialized module 'erfa' (most likely due to a circular import) (C:\Users\kbagnall\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\erfa\__init__.py)

**Comment by mhvk** [comment]
@kbagnall-quera - I'm sorry that you never got any response to your issue. Has it been resolved in the meantime, or do you still have trouble?

**Comment by doronbehar** [comment]
I'm experiencing the same issue here when trying to enable tests during the build of this package in NixOS. Here's my full log:

https://gist.github.com/doronbehar/d1a8b891a8be41b869283b3d17a92388

I sort of managed to fix this by removing the `erfa/` directory before running the check, but it then fails as explained in #112.

**Comment by doronbehar** [comment]
Interesting.. Now I learned that if I run `python -m pytest` I get the failures as before, but if I simply run `pytest` I don't get them (but I still hit #112). This may be something specific to our build system... Continuing investigating.

**Comment by mhvk** [comment]
@doronbehar - as I noted in #112, removing `erfa` just means removing doing most of the tests, so that is no solution! Your issue also seems rather different from that initially reported here, in that somewhere inside `erfa` itself the import could wrong. Would it be possible to open a new issue? It would be helpful if you could describe in more detail what you did to build the package, what versions of the various components you are on, etc.

**Comment by doronbehar** [comment]
Hey and thanks for the quick reply :) As I tried to write in my previous comment, the issue I experienced that was similar to this one eventually was solved by using `pytest` directly, and not use `python -m pytest`. I no longer experience this issue, and I no longer remove my `erfa` directory. I described my investigation regarding this in detail [here](https://github.com/NixOS/nixpkgs/issues/255262), and I don't want to put it on your responsibility, this is a Nix specific issue.
